pom.xml

```diff
@@ -242,6 +247,7 @@
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
             <scope>test</scope>
+            <version>${junit.version}</version>
```

Comment:
I didn't realize we have both Junit4 & Junit5 on the classpath. We should only have Junit5, as it has backwards compatibility features for Junit4 and Junit3.

Reply:
The problem is that we are using a method that is not present without it like `assertNotEquals`.

Reply:
Yeah, but as I said, there are [backwards compatibility features](https://stackoverflow.com/questions/41635164/is-junit5-backward-compatible-to-junit4) that can be used instead of having both engines on the classpath. But it seems unrelated with the main issue, so can go to a separate PR.